data/src/main/java/com/microsoft/azure/kusto/data/ObjectPopulator.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.azure.kusto.data;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Represents a single Kusto result column and its mapping to a java object. Used to extract the column from a {@link KustoResultSetTable} and set the
+ * value in a pojo. The column is defined by either its name or its ordinal (or both, in which case the ordinal is used).
+ * 
+ * @param <R>
+ *            the pojo this column needs to set its value to
+ * 			
+ * @param <C>
+ *            the Java class returned by the ADX column type
+ * @param <T>
+ *            type of ADX column
+ */
+class ObjectPopulator<R, C, T extends KustoType<C>> {
+	
+	final static int UNSET_ORDINAL = -1;
+	
+	final BiConsumer<R, C> valueSetter;
+	final String name;
+	final T type;
+	final boolean isNullable;
+	final int presetOrdinal;
+	
+	static <R, C, T extends KustoType<C>> ObjectPopulator<R, C, T> of(String name, int ordinal, T type, boolean isNullable, BiConsumer<R, C> valueSetter) {
+		return new ObjectPopulator<>(name, ordinal, type, isNullable, valueSetter);
+	}
+	
+	static <R, C, T extends KustoType<C>> ObjectPopulator<R, C, T> of(String name, T type, boolean isNullable, BiConsumer<R, C> valueSetter) {
+		return new ObjectPopulator<>(name, UNSET_ORDINAL, type, isNullable, valueSetter);
+	}
+	
+	static <R, C, T extends KustoType<C>> ObjectPopulator<R, C, T> of(int ordinal, T type, boolean isNullable, BiConsumer<R, C> valueSetter) {
+		return new ObjectPopulator<>(null, ordinal, type, isNullable, valueSetter);
+	}
+	
+	ObjectPopulator(String name, int ordinal, T type, boolean isNullable, BiConsumer<R, C> valueSetter) {
+		this.name = name;
+		this.presetOrdinal = ordinal;
+		this.type = type;
+		this.isNullable = isNullable;
+		this.valueSetter = valueSetter;
+	}
+	
+	void populateFrom(R objToPopulate, KustoResultSetTable resultSet) {
+		populateFrom(objToPopulate, resultSet, this.presetOrdinal);
+	}
+	
+	void populateFrom(R objToPopulate, KustoResultSetTable resultSet, int ordinal) {
+		Object o = resultSet.getObject(columnIndexInResultSet(resultSet));
```

Comment:
It seems calling columnIndexInResultSet() here is an unnecessary duplicate call. In the flow coming from KustoResultMapper.extractList(), we already executed this method and passed it in as the "ordinal" parameter.

Reply:
Right, I probably mixed it with the flow from `extractSingle()`. It turns out I wasn't actually using the ordinal passed to `populateFrom` at all except for in the exception text. Fixed it now, please take a look.